### DISCUSSION TITLE
DKI reconstruction for in-vivo mice brain (black voxels)

### DISCUSSION THREAD

**Discussion by HopperLiu** [dipy/dipy#3334] [Reconstruction]
Hi there,

I recently meet a problem in DKI reconstructions. There are lots of black voxel.
I have applied gibbs_removal and denoised. The reconstruction algorithm is 'WLS'.
What should I do to improve my results?
Thank you very much for your assistance.
AK:
![3](https://github.com/user-attachments/assets/c19ba448-0735-4314-a992-c0c42ebbdc50)
FKA:
![2](https://github.com/user-attachments/assets/e3986c4c-0b4f-47a8-98e1-2d296d2b05d6)
MK:
![4](https://github.com/user-attachments/assets/3f20be4b-0dd2-43d8-b89d-660ac5e6cfbc)
MKT:
![5](https://github.com/user-attachments/assets/20deb6ad-7877-4af5-9d5c-d30b7a195cba)
RK:
![6](https://github.com/user-attachments/assets/6a28a75e-d9a6-4156-91a3-4af3f952e118)
RTK:
![7](https://github.com/user-attachments/assets/44ddca23-7955-454d-9eba-619081c6ca1f)

**Comment by skoudoro** [comment]
Implausible negative kurtosis estimates,  creates 'black' voxels or holes in DKI maps. These artifacts can compromise the accuracy of the DKI analysis, and many solutions are possible to ensure more reliable results.

I recommend to read the last version of our tutorials on this: https://github.com/dipy/dipy/blob/master/doc/examples/reconst_dki.py

I hope it helps

**Reply by skoudoro** [reply]
especially from this line: https://github.com/dipy/dipy/blob/8be51800309373135cdda26bbb7bfd9d310c1d41/doc/examples/reconst_dki.py#L243

**Comment by samcoveney** [comment]
You could also use the code I added to do robust DKI estimates, with / without convexity constraints: https://www.biorxiv.org/content/10.1101/2025.02.12.637822v1

Tutorial is still pending but feel free to contact me
